lib_eio_luv/eio_luv.ml

```diff
@@ -126,6 +126,26 @@ let await_with_cancel ~request fn =
         )
     )
 
+let await_with_cancel_poll ~poll fn =
+  enter (fun st k ->
+    let cancel_reason = ref None in
+    Eio.Private.Fibre_context.set_cancel_fn k.fibre (fun ex ->
+        cancel_reason := Some ex;
+        match Luv.Poll.stop poll with
+        | Ok () -> ()
+        | Error e -> Log.debug (fun f -> f "Cancel failed: %s" (Luv.Error.strerror e))
+      );
+    fn st.loop (fun v ->
```

Comment:
From http://docs.libuv.org/en/v1.x/poll.html#c.uv_poll_start it looks like it will keep calling the callback as long as the condition is true. I suspect you need to stop polling here, but I haven't tested it.